### ISSUE TITLE
Allow BLE and BT Classic integrations to run from the same adapter

### ISSUE THREAD

**Issue by mKeRix** [mKeRix/room-assistant#209]
**Is your feature request related to a problem? Please describe.**
Currently, you can only run BLE or BT Classic integrations on one hci device at the same time. USB Bluetooth sticks are generally cheap to come by, but a bit annoying to stick into the Pi Zeros. It would be much nicer if we could mix any integrations using these two different communication modes and use them from the same HCI device.

**Describe the solution you'd like**
room-assistant should prioritize Bluetooth device usage and start/stop BLE scanning as well as Bluetooth queries automatically so that the adapter can be shared. This should happen transparently for the user and also be able to handle multiple adapters for more performance.

**Describe alternatives you've considered**
Buying USB Bluetooth sticks for every machine and then using the `hciDeviceId` options accordingly.

**Additional context**
From a technical standpoint it makes sense to structure out the Bluetooth device code into a separate Bluetooth module which handles all the hardware stuff. The other integrations could then just declare that module as dependency and hook into the newly provided APIs.

**Comment by ghzgod** [comment]
How much of a donation do we need to contribute to get this working?  I bought a second BT adapter for every device I have and now I am stricken with this issue.  

https://www.raspberrypi.org/forums/viewtopic.php?t=282948

I would rather have it all running on the same device.  I can write the code if you just tell me what I need to look at.

**Comment by mKeRix** [comment]
It's more an issue of time than the amount of donations. I would love to see this working as well, since I think I would also get some use out of this. I just haven't gotten around to working on this yet.

My idea when writing this ticket was to bundle the logic that uses the Bluetooth adapter resources in a separate component, which can then be inserted as dependency of both the Bluetooth Classic and the Bluetooth Low Energy integrations. Inside the new component we could have a priority system, which turns off BLE scanning when a Classic inquiry is being done and turns it back on after (depending on what is loaded of course). Additionally we would need to manage the case of having multiple adapters in that component - that could maybe be done quite easily via the NestJS dependency injection features, but I'm not sure.

**Comment by ghzgod** [comment]
Incredible.  You have been busy today!  Thank you so much dev! This is a game changer!!!!

**Comment by mKeRix** [comment]
@ghzgod Let me know if you run into issues while testing the beta version. I will also leave it running together with the Xiaomi integration in my setup for a while to ensure this is working & stable. :)

**Comment by ghzgod** [comment]
No problem, if you need help anywhere else let me know!  I am updating my pi's now and I'll make my config use the same HCI dev #

**Comment by ghzgod** [comment]
Also it is awesome you live in Berlin.  I am not too far from you.  I was just there before all this COVID stuff happened.

**Comment by ghzgod** [comment]
So I ran `sudo npm i --global --unsafe-perm room-assistant@beta`on my RPI and upon run I received 

`/opt/nodejs/lib/node_modules/room-assistant/node_modules/command-line-args/dist/index.js:1345
        throw err`

Not too sure how to resolve this.

**Comment by mKeRix** [comment]
Upon running the install or room-assistant itself?

The library that is throwing this error wasn't updated, so everything should work as before in theory. Could you maybe try installing room-assistant a second time like you did above and then run it again? And if that does not work: could you post the command you use for running it and which NodeJS version you are currently on?

**Comment by ghzgod** [comment]
`nmp -v` = "6.14.6"
'node -v' = "v10.22.1"

I run  `sudo npm i --global --unsafe-perm room-assistant@beta` to update ra to the beta version.

When I run `room-assistant ~/room-assistant/config/local.yml` I get that throw err posted above.  Full error below:

`/opt/nodejs/lib/node_modules/room-assistant/node_modules/command-line-args/dist/index.js:1345
        throw err
        ^

UNKNOWN_VALUE: Unknown value: /home/pi/room-assistant/config/local.yml
    at commandLineArgs (/opt/nodejs/lib/node_modules/room-assistant/node_modules/command-line-args/dist/index.js:1342:21)
    at Object.<anonymous> (/opt/nodejs/lib/node_modules/room-assistant/bin/room-assistant.js:64:17)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)`

**Comment by mKeRix** [comment]
You need to run room-assistant like this: `room-assistant -c ~/room-assistant/config/`. The CLI options are documented [here](https://www.room-assistant.io/guide/cli.html).

**Comment by ghzgod** [comment]
Ah interesting and thank you.  The command I posted above has worked for me before but then I wonder if it was running and ignoring the path supplied because I didn't put -c.  It is running now!  I switched both HCI Device numbers to the USB BT adapter I have.  

I notice an occasional error about 10X at once and then not again until my next scan:

`processLeAdvertisingReport: Caught illegal packet (buffer overflow): RangeError [ERR_BUFFER_OUT_OF_BOUNDS]: Attempt to write outside buffer bounds`

However it is properly tracking my devices both BT Classic and BLE from the same adapter!  Anything else you would like me to test?
